File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/KnowledgeBaseCreateRequest.kt
package com.sdkwork.ai

data class KnowledgeBaseCreateRequest(
    val name: String? = null,
    val description: String? = null,
    val extra: Map<String, Any>? = null,
    val embeddingModelId: String? = null,
    val indexType: String? = null
)

File: sdkwork-ai-sdk-csharp/Models/KnowledgeBaseCreateRequest.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Ai.Models
{
    public class KnowledgeBaseCreateRequest
    {
        public string? Name { get; set; }
        public string? Description { get; set; }
        public Dictionary<string, object>? Extra { get; set; }
        public string? EmbeddingModelId { get; set; }
        public string? IndexType { get; set; }
    }
}
